turbopack/crates/turbopack-ecmascript/src/analyzer/graph.rs

```diff
@@ -965,13 +1004,8 @@ impl Analyzer<'_> {
                 arrow_expr,
                 ArrowExprField::Params(i),
             ));
-            if let Some(arg) = iter.next() {
-                self.current_value = Some(self.eval_context.eval(&arg.expr));
-                self.visit_pat(param, &mut ast_path);
-                self.current_value = None;
```

Comment:
so at certain syntactic points we actually expect current_value to be cleared.

should there be an alternative `with_pat_value` that basically just asserts that we are starting and returning to `None`?

Reply:
I suppose I don't think there is a bug now, and perhaps this new pattern makes a potential bug around accidentally resetting to `None` impossible.  but surely there are points where we expect to reset to `none`? shoudl we maintain that?

Reply:
Yeah, my thought was that all the places where we explicitly cleared `pat_value` were one of two reasons:

- We know that the pat value is always `None` in that parent node, and so there's no point in saving/restoring it, which would be messier to do syntactically. In theory, we're now paying some tiny extra performance cost here by needing to move this `None` value onto the stack.

- To avoid potential bugs caused by sibling nodes that fail to clear it, which is now impossible to forget to do.

We do have `take_pat_value` which does reset `pat_value` to `None`, but just that's an optimization to avoid clones, rather than as a way of explicitly clearing `pat_value`.